test/e2e/host_path.go

```diff
@@ -0,0 +1,128 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/latest"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"os"
+	"path"
+
+	. "github.com/onsi/ginkgo"
+)
+
+//TODO : Consolodate this code with the code for emptyDir.
+//This will require some smart.
+var _ = Describe("hostDir", func() {
+	var (
+		c *client.Client
+	)
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		expectNoError(err)
+
+		//cleanup before running the test.
+		_ = os.Remove("/tmp/test-file")
+	})
+
+	It("MOD volume on tmpfs should have the correct mode", func() {
+		volumePath := "/test-volume"
+		source := &api.HostPathVolumeSource{
+			Path: "/tmp",
+		}
+		pod := testPodWithHostVol(volumePath, source)
+
+		pod.Spec.Containers[0].Args = []string{
+			fmt.Sprintf("--fs_type=%v", volumePath),
+			fmt.Sprintf("--file_mode=%v", volumePath),
+		}
+		testContainerOutput("emptydir r/w on tmpfs", c, pod, []string{
+			"mount type of \"/test-volume\": tmpfs",
+			"mode of file \"/test-volume\": dtrwxrwxrwx", // we expect the sticky bit (mode flag t) to be set for the dir
+		})
+	})
+
+	It("should support r/w on tmpfs", func() {
+		volumePath := "/test-volume"
+		filePath := path.Join(volumePath, "test-file")
+		source := &api.HostPathVolumeSource{
+			Path: "/tmp",
+		}
+		pod := testPodWithHostVol(volumePath, source)
+
+		pod.Spec.Containers[0].Args = []string{
+			fmt.Sprintf("--fs_type=%v", volumePath),
+			fmt.Sprintf("--rw_new_file=%v", filePath),
+			fmt.Sprintf("--file_mode=%v", filePath),
+		}
+		testContainerOutput("emptydir r/w on tmpfs", c, pod, []string{
+			"mount type of \"/test-volume\": tmpfs",
+			"mode of file \"/test-volume/test-file\": -rw-r--r--",
+			"content of file \"/test-volume/test-file\": mount-tester new file",
+		})
+	})
+})
+
+//These constants are borrowed from the other test.
+//const containerName = "test-container"
+//const volumeName = "test-volume"
+
+func mount(source *api.HostPathVolumeSource) []api.Volume {
+	return []api.Volume{
+		{
+			Name: volumeName,
+			VolumeSource: api.VolumeSource{
+				HostPath: source,
+			},
+		},
+	}
+}
+
+//TODO: To merge this with the emptyDir tests, we can make source a lambda.
+func testPodWithHostVol(path string, source *api.HostPathVolumeSource) *api.Pod {
+	podName := "pod-" + string(util.NewUUID())
+
+	return &api.Pod{
+		TypeMeta: api.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: latest.Version,
+		},
+		ObjectMeta: api.ObjectMeta{
+			Name: podName,
+		},
+		Spec: api.PodSpec{
+			Containers: []api.Container{
+				{
+					Name:  containerName,
+					Image: "kubernetes/mounttest:0.1",
```

Comment:
Question: Do we need this image? Instead, we can run a few bash commands that would achieve the desired goals.
Not depending on a custom image for testing makes it easy to update and maintain.